File: java/src/patterns/creational/factory/concretecreators/XmlParserCreator.java
package patterns.creational.factory.concretecreators;

import patterns.creational.factory.abstractcreator.ParserCreator;
import patterns.creational.factory.concreteproducts.XmlParser;
import patterns.creational.factory.productinterface.DocumentParser;

public class XmlParserCreator extends ParserCreator {
    @Override
    public DocumentParser createParser() {
        return new XmlParser();
    }
}

File: python/src/patterns/creational/factory/concrete_creators/xml_parser_creator.py
from src.patterns.creational.factory.abstract_creator.parser_creator import ParserCreator
from src.patterns.creational.factory.concrete_products.xml_parser import XmlParser
from src.patterns.creational.factory.product_interface.document_parser import DocumentParser

class XmlParserCreator(ParserCreator):
    def create_parser(self) -> DocumentParser:
        return XmlParser()
